cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,150 @@
+# CLI commands for User-Defined Types
+
+* **Author**: Reshma Abdul Rahim (reshrahim)
+
+## Topic Summary
+<!-- A paragraph or two to summarize the topic area. Just define it in summary form so we all know what it is. -->
+User-Defined Types enable Platform engineers to define and deploy their organization's custom services in Radius. More details around the feature, user scenarios and experience can be found [here](/architecture/2024-06-resource-extensibility-feature-spec.md). This document will focus on the CLI experience for UDTs.
+
+### Top level goals
+<!-- At the most basic level, what are we trying to accomplish? -->
+- Provide an intuitive and seamless experience for platform engineers to create, update, and delete UDTs using the CLI.
+- Ensure that the CLI experience is consistent with the other commands in Radius
+
+### Non-goals (out of scope)
+<!-- What are we explicitly not trying to accomplish? -->
+- Other interfaces and experiences to interact with UDTs are out of scope for this document
+
+## Key scenarios
+<!-- List ~3-7 high level scenarios to clarify the value and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->
+The key scenarios for UDTs are outlined [here](/architecture/2024-06-resource-extensibility-feature-spec.md). We will decompose these scenarios specific to the CLI experience.
+
+### Scenario 1: Deb defines and deploys the internal messaging service using the CLI
+<!-- One or two sentence summary -->
+Deb, a platform engineer who wants to define and deploy their internal messaging service Plaid as a resource type in Radius. 
+
+Before we start designing the experience, we need to understand the high level terminologies/concept that a user would interact with in the CLI experience.
+
+| Terminology | Description | Example |
+|-------------|-------------|---------|
+|Resource provider | Refers to a group of resource types under a common namespace. Resource providers are also entities that implement the group of resource types | Mycompany.Messaging or Applications.Core |
+|Resource type | A service or technology that can be modeled as a resource | PostgreSQL or AWS S3 or Internal messaging service |
+|Resource | An instance of a resource type | mydb or mybucket |
+
+#### User Experience
+
+1. Deb uses the rad CLI to scaffold a `yaml` template definition for the internal messaging service `Plaid`
```

Comment:
Is this step really necessary? I'd prefer that authoring a resource type is so simple that it does not require a template. I'm imagining if I want to create a mycompany.com/database that I only need a single YAML file. If that's the case, I can just copy/paste from the sample in the docs.

If the scaffolding is indeed creating multiple files, I think we should challenge ourselves to simplify that further.

Reply:
Yes I think its helps user get a schema template with basic properties of the resource type from a rad command. Its convenience for the user over manually creating and copy pasting the template from docs. 

We do this already to scaffold an application(app.bicep) when you do `rad init`

Reply:
Ok. I will move this to commands to consider for future iterations